Intermittent test failure in blocklist/tests/test_mlbf.py: test_generate_and_write_mlbf()

We've seen this a few times:
```
____________________ TestMLBF.test_generate_and_write_mlbf _____________________
[gw0] linux -- Python 3.6.7 /home/travis/build/mozilla/addons-server/.tox/main/bin/python3
src/olympia/blocklist/tests/test_mlbf.py:274: in test_generate_and_write_mlbf
    assert bfilter.bitCount() == 1384
E   AssertionError: assert 2824 == 1384
E     +2824
E     -1384
        bfilter    = <filtercascade.FilterCascade object at 0x7f5731cec358>
        buffer     = (b'\x02\x00\x00\x10\xaa\x8f^\\r4\xf9\x9a%%\xb4\x0eI\x1e\xdf\x89\x02h\x05\x00'
 b'\x00\x01\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x80\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00 \x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x10\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00 \x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x02\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x04\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00 \x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x04\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x02\xa0\x05\x00\x00\x01\x00\x00\x00\x02\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x02\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00')
        filter_file = <_io.BufferedReader name='/tmp/pytest-of-travis/pytest-0/popen-gw0/test_generate_and_write_mlbf0/storage/mlbf/123456/filter'>
        mlbf       = <olympia.blocklist.mlbf.MLBF object at 0x7f57316b94a8>
        self       = <olympia.blocklist.tests.test_mlbf.TestMLBF testMethod=test_generate_and_write_mlbf>
```

--- a/src/olympia/blocklist/tests/test_mlbf.py
+++ b/src/olympia/blocklist/tests/test_mlbf.py
@@ -287,7 +287,7 @@ class TestMLBF(TestCase):
                 continue
             key = mlbf.KEY_FORMAT.format(guid=guid, version=version_str)
             assert key not in bfilter
-            
+
         assert os.stat(mlbf.filter_path).st_size == 203, (
             blocked_guids, all_addons)
         assert bfilter.bitCount() == 1384, (

--- a/src/olympia/blocklist/tests/test_mlbf.py
+++ b/src/olympia/blocklist/tests/test_mlbf.py
@@ -274,20 +274,24 @@ class TestMLBF(TestCase):
             buffer = filter_file.read()
             bfilter = FilterCascade.from_buf(buffer)
 
-        assert bfilter.bitCount() == 1384
         blocked_versions = mlbf.fetch_blocked_from_db()
-        for guid, version_str in blocked_versions.values():
+        blocked_guids = blocked_versions.values()
+        for guid, version_str in blocked_guids:
             key = mlbf.KEY_FORMAT.format(guid=guid, version=version_str)
             assert key in bfilter
 
         all_addons = mlbf.fetch_all_versions_from_db(blocked_versions.keys())
         for guid, version_str in all_addons:
             # edge case where a version_str exists in both
-            if (guid, version_str) in blocked_versions.values():
+            if (guid, version_str) in blocked_guids:
                 continue
             key = mlbf.KEY_FORMAT.format(guid=guid, version=version_str)
             assert key not in bfilter
-        assert os.stat(mlbf.filter_path).st_size == 203
+            
+        assert os.stat(mlbf.filter_path).st_size == 203, (
+            blocked_guids, all_addons)
+        assert bfilter.bitCount() == 1384, (
+            blocked_guids, all_addons)
 
     def test_generate_diffs(self):
         old_versions = [

